File: src/main/java/rak/planetGenerator/parsers/BiomeParser.java
package rak.planetGenerator.parsers;

import java.io.IOException;
import java.io.InputStream;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;

import javafx.scene.paint.Color;
import rak.planetGenerator.PlanetGeneratorApplication;
import rak.planetGenerator.model.Biome;
import rak.planetGenerator.model.BiomeAttributes;
import rak.planetGenerator.model.BiomeCollection;

public class BiomeParser {
	
	public BiomeCollection parseBiomes(String biomeFileName){
		BiomeCollection biomes = new BiomeCollection();
		
		ArrayNode root = null;
		try {
			root = parseBiomesJson(biomeFileName);
			generateBiomes(root, biomes);
		} catch (IOException e) {
			e.printStackTrace();
		}
		
		return biomes;
	}

	private ArrayNode parseBiomesJson(String biomeFileName) throws IOException, JsonProcessingException {
		String name = "json/" + biomeFileName +".json";
//		InputStream in = getClass().getResourceAsStream(name);
		InputStream in = PlanetGeneratorApplication.getResourceAsStream(name);
		ObjectMapper mapper = new ObjectMapper();
		ArrayNode root = (ArrayNode) mapper.readTree(in);
		return root;
	}
	
	private void generateBiomes(ArrayNode root, BiomeCollection biomes) {
		for (JsonNode biomeJson : root){
			BiomeAttributes biome = parseBiomeAndAttributes(biomeJson);
			biomes.addItem(biome);
		}
	}

	private BiomeAttributes parseBiomeAndAttributes(JsonNode biomeJson) {
		Biome biome = parseBiome(biomeJson);
		
		BiomeAttributes biomeAttributes = new BiomeAttributes(biome);
		
		return biomeAttributes;
	}


	private Biome parseBiome(JsonNode biomeJson) {
		String id = biomeJson.get("id").textValue();
		String name = biomeJson.get("name").textValue();
		
		int altitude = biomeJson.get("altitude").intValue();
		int temperature = biomeJson.get("temperature").intValue();
		int percipitation = biomeJson.get("percipitation").intValue();
		
		int altitudeVariation = biomeJson.get("altitudeMaxVariance").intValue();
		int temperatureVariation = biomeJson.get("temperatureMaxVariance").intValue();
		int percipitationVariation = biomeJson.get("percipitationMaxVariance").intValue();
		
		String colorHex = biomeJson.get("color").textValue();
		Color color = Color.valueOf(colorHex);
		
		Biome biome = new Biome(id, name);
		biome.setColor(color);
		biome.setAltitude(altitude);
		biome.setTemperature(temperature);
		biome.setPercipitation(percipitation);
		
		biome.setAltitudeVariation(altitudeVariation);
		biome.setTemperatureVariation(temperatureVariation);
		biome.setPercipitationVariation(percipitationVariation);
		
		return biome;
	}
	

	
}

File: src/main/java/rak/planetGenerator/parsers/BiomeParser.kt
package rak.planetGenerator.parsers

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.node.ArrayNode
import javafx.scene.paint.Color
import rak.planetGenerator.model.Biome
import rak.planetGenerator.model.BiomeAttributes
import rak.planetGenerator.model.BiomeCollection
import rak.utility.ResourceLoader
import java.io.IOException

class BiomeParser {

    fun parseBiomes(biomeFileName: String): BiomeCollection {
        val biomes = BiomeCollection()

        try {
            val root = parseBiomesJson(biomeFileName)
            generateBiomes(root, biomes)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return biomes
    }

    @Throws(IOException::class, JsonProcessingException::class)
    private fun parseBiomesJson(biomeFileName: String): ArrayNode {
        val name = "json/$biomeFileName.json"
        //		InputStream in = getClass().getResourceAsStream(name);
        val `in` = ResourceLoader.getResourceAsStream(name)
        val mapper = ObjectMapper()
        return mapper.readTree(`in`) as ArrayNode
    }

    private fun generateBiomes(root: ArrayNode, biomes: BiomeCollection) {
        for (biomeJson in root) {
            val biome = parseBiomeAndAttributes(biomeJson)
            biomes.addItem(biome)
        }
    }

    private fun parseBiomeAndAttributes(biomeJson: JsonNode): BiomeAttributes {
        val biome = parseBiome(biomeJson)

        return BiomeAttributes(biome)
    }


    private fun parseBiome(biomeJson: JsonNode): Biome {
        val id = biomeJson.get("id").textValue()
        val name = biomeJson.get("name").textValue()

        val altitude = biomeJson.get("altitude").intValue()
        val temperature = biomeJson.get("temperature").intValue()
        val precipitation = biomeJson.get("percipitation").intValue()

        val altitudeVariation = biomeJson.get("altitudeMaxVariance").intValue()
        val temperatureVariation = biomeJson.get("temperatureMaxVariance").intValue()
        val precipitationVariation = biomeJson.get("percipitationMaxVariance").intValue()

        val colorHex = biomeJson.get("color").textValue()
        val color = Color.valueOf(colorHex)

        val biome = Biome(id, name)
        biome.color = color
        biome.altitude = altitude
        biome.temperature = temperature
        biome.percipitation = precipitation

        biome.altitudeVariation = altitudeVariation
        biome.temperatureVariation = temperatureVariation
        biome.percipitationVariation = precipitationVariation

        return biome
    }


}
